services/venice-controller/src/main/java/com/linkedin/venice/controller/server/ClusterAdminOpsRequestHandler.java

```diff
@@ -148,4 +149,30 @@ public AdminTopicMetadataGrpcResponse updateAdminTopicMetadata(UpdateAdminTopicM
         AdminTopicMetadataGrpcResponse.newBuilder().setMetadata(adminTopicGrpcMetadataBuilder.build());
     return responseBuilder.build();
   }
+
+  public AdminTopicMetadataGrpcResponse updateAdminOperationProtocolVersion(
+      UpdateAdminOperationProtocolVersionGrpcRequest request) {
+    String clusterName = request.getClusterName();
+    if (StringUtils.isBlank(clusterName)) {
+      throw new IllegalArgumentException("Cluster name is required for updating admin operation protocol version");
+    }
+
+    long adminOperationProtocolVersion = request.getAdminOperationProtocolVersion();
+    if (adminOperationProtocolVersion == 0 || adminOperationProtocolVersion < -1) {
+      throw new IllegalArgumentException(
+          "Admin operation protocol version is required and must be -1 or greater than 0");
+    }
+
+    LOGGER.info(
+        "Updating admin operation protocol version for cluster: {} to version: {}",
+        clusterName,
+        adminOperationProtocolVersion);
+
+    admin.updateAdminOperationProtocolVersion(clusterName, adminOperationProtocolVersion);
+
+    AdminTopicGrpcMetadata.Builder adminMetadataBuilder = AdminTopicGrpcMetadata.newBuilder()
+        .setClusterName(clusterName)
+        .setAdminOperationProtocolVersion(adminOperationProtocolVersion);
+    return AdminTopicMetadataGrpcResponse.newBuilder().setMetadata(adminMetadataBuilder.build()).build();
```

Comment:
Is it the common way to write gRpc? I am curious what's reasoning behind building yet another metadata instance. Is it for improving modularization and readability? cc: @sushantmane

Reply:
> Is it for improving modularization 
Generally, yes!

I think @minhmo1620 is trying to reuse AdminTopicMetadataGrpcResponse for this endpoint. The response includes an inner field, AdminTopicGrpcMetadata, which was separated to enable reuse across other APIs.

Reply:
I was trying to reuse `AdminTopicGrpcMetadata` so we don't need to duplicate the available fields inside metadata multiple times. The metadata is compacted, which I think is cleaner. 

In case you ask why don't we bring it out, let's say return `AdminTopicGrpcMetadata` directly instead of `AdminTopicMetadataGrpcResponse`. We can do that, however, the benefit of the current setup is clearing out the request and response object. In case we want to extend the current response object to have something else (that not part of metadata), we can easily add it in the same level with the metadata. I don't have strong preference here, I'm open for discussion on this!

Reply:
The current setup is okay. Thanks!